Description: Some additional adaptions to libvirt-guests
 Add some additional code which sets LIBVIRT_DEFAULT_URI if running in
 a Xen control domain. The built-in detection does not really work because
 qemu (not accelerated/KVM) is is still available as conntection driver.
 Though very unlikely desired as the default then.
 Modify the default config to do 10 parallel shutdown requests and reduce
 the timeout to 120s/2m.
Forwarded: no
Author: Stefan Bader <stefan.bader@canonical.com>

Index: libvirt-1.3.1/tools/libvirt-guests.sh.in
===================================================================
--- libvirt-1.3.1.orig/tools/libvirt-guests.sh.in	2016-03-02 11:22:16.413519687 +0100
+++ libvirt-1.3.1/tools/libvirt-guests.sh.in	2016-03-04 14:41:00.806625419 +0100
@@ -61,6 +61,15 @@ VAR_SUBSYS_LIBVIRT_GUESTS="$localstatedi
 
 RETVAL=0
 
+# Default URI is not correct in the Xen case as the non-accelerated
+# qemu driver also gets initialized.
+if [ -f "/proc/xen/capabilities" ]; then
+    if [ "$(cat /proc/xen/capabilities)" = "control_d" ]; then
+        LIBVIRT_DEFAULT_URI="xen:///"
+        export LIBVIRT_DEFAULT_URI
+    fi
+fi
+
 # retval COMMAND ARGUMENTS...
 # run command with arguments and convert non-zero return value to 1 and set
 # the global return variable
Index: libvirt-1.3.1/tools/libvirt-guests.sysconf
===================================================================
--- libvirt-1.3.1.orig/tools/libvirt-guests.sysconf	2016-03-01 10:52:53.000000000 +0100
+++ libvirt-1.3.1/tools/libvirt-guests.sysconf	2016-03-04 14:44:35.802622104 +0100
@@ -26,14 +26,14 @@
 
 # If set to non-zero, shutdown will suspend guests concurrently. Number of
 # guests on shutdown at any time will not exceed number set in this variable.
-#PARALLEL_SHUTDOWN=0
+PARALLEL_SHUTDOWN=10
 
 # Number of seconds we're willing to wait for a guest to shut down. If parallel
 # shutdown is enabled, this timeout applies as a timeout for shutting down all
 # guests on a single URI defined in the variable URIS. If this is 0, then there
 # is no time out (use with caution, as guests might not respond to a shutdown
 # request). The default value is 300 seconds (5 minutes).
-#SHUTDOWN_TIMEOUT=300
+SHUTDOWN_TIMEOUT=120
 
 # If non-zero, try to bypass the file system cache when saving and
 # restoring guests, even though this may give slower operation for
